### PR TITLE
Bump acorn 8

### PR DESCRIPTION
**Pull request by JLHwung**
Bump Acorn to 8 per requested in https://github.com/webpack/webpack/pull/11550#issuecomment-700698799

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
Yes, finally.

**Does this PR introduce a breaking change?**
No. The breaking change from acorn 7 to acorn 8 is that `ecmaVersion` is a required option.

**What needs to be documented once your changes are merged?**
No.

### CODE REVIEW

File: test/cases/parsing/logical-assignment/test.filter.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var supportsLogicalAssignment = require("../../../helpers/supportsLogicalAssignment");
```
Comment (sokra) [inline test/cases/parsing/logical-assignment/test.filter.js:1]: You forgot to add this file

**Review by sokra — CHANGES_REQUESTED** [review]
Something seems pretty broken? CI is complaining a lot. I wonder why?

**Comment by JLHwung** [comment]
Please restart the CI, I believe the errors are not related.

**Comment by JLHwung** [comment]
Ahh the CI is not flaky. It bails because of an upstream issue: https://github.com/acornjs/acorn/issues/981

**Comment by sokra** [comment]
Thanks

**Pull request merged**